signalfx/ingest.py

```diff
@@ -271,31 +273,55 @@ def _set_event_properties(self, pbuf_obj, properties):
             prop.key = key
             self._assign_property_value(prop, value)
 
-    def _assign_property_value(self, prop, value):
-        if isinstance(value, int):
-            prop.value.intValue = value
-        elif isinstance(value, str):
-            prop.value.strValue = value
-        elif isinstance(value, unicode):
-            prop.value.strValue = value
-        elif isinstance(value, float):
-            prop.value.doubleValue = value
-        elif isinstance(value, bool):
-            prop.value.boolValue = value
-        else:
-            raise ValueError('Invalid Value ' + str(value))
-
-    def _assign_value_type(self, pbuf_dp, value):
-        if isinstance(value, int):
-            pbuf_dp.value.intValue = value
-        elif isinstance(value, str):
-            pbuf_dp.value.strValue = value
-        elif isinstance(value, unicode):
-            pbuf_dp.value.strValue = value
+    def _assign_value_by_type(self, pbuf_obj, value, _bool=True, _float=True,
+                        _integer=True, _string=True):
+        """Assigns the supplied value to the appropriate protobuf value type"""
+        # bool inherits int, so bool instance check must be executed prior to
+        # checking for integer types
+        if isinstance(value, bool):
+            if _bool is True:
```

Comment:
I condensed the integer range checks into a single statement.  But, by having the extra if/else for _integer, _float.. etc we can provide an appropriate error message if someone tries to use an explicitly unsupported type.  The problem is that when combining the original functionality of _assign_value and _assign_property_value determining what is and isn't a supported type changes based on who invokes _assign_value_by_type.  The secondary if/else lets me tell the user that they supplied a bool for their metric value, but bool is an unsupported type.  The alternative is the customer sees "ValueError: Invalid value <value>" regardless of the offending type they supply.